cluster-provision/gocli/providers/provision.go

```diff
@@ -0,0 +1,214 @@
+		return err
+	}
+
+	// Wait for ssh.sh script to exist
+	_, err = docker.Exec(kp.Docker, kp.nodeContainer(kp.Version, nodeName), []string{"bin/bash", "-c", "while [ ! -f /ssh_ready ] ; do sleep 1; done", "checking for ssh.sh script"}, os.Stdout)
+	if err != nil {
+		return err
+	}
+
+	sshClient, err := libssh.NewSSHClient(sshPort, 1, false)
+	if err != nil {
+		return err
+	}
+
+	rootkey := rootkey.NewRootKey(sshClient)
+	if err = rootkey.Exec(); err != nil {
+		fmt.Println(err)
+	}
+
+	sshClient, err = libssh.NewSSHClient(sshPort, 1, true)
+	if err != nil {
+		return err
+	}
+
+	if strings.Contains(kp.Phases, "linux") {
+		provisionOpt := provisionopt.NewLinuxProvisioner(sshClient)
+		if err = provisionOpt.Exec(); err != nil {
+			return err
+		}
+	}
+
+	if strings.Contains(kp.Phases, "k8s") {
+		// copy provider scripts
+		if err = sshClient.Command("mkdir -p /tmp/ceph /tmp/cnao /tmp/nfs-csi /tmp/nodeports /tmp/prometheus /tmp/whereabouts /tmp/kwok"); err != nil {
+			return err
+		}
+		// Copy manifests to the VM
+		success, err := docker.Exec(kp.Docker, kp.nodeContainer(kp.Version, nodeName), []string{"/bin/bash", "-c", "scp -r -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -i vagrant.key -P 22 /scripts/manifests/* root@192.168.66.101:/tmp"}, os.Stdout)
+		if err != nil {
+			return err
+		}
+
+		if !success {
+			return fmt.Errorf("error copying manifests to node")
+		}
+
+		provisionK8sOpt := k8sprovision.NewK8sProvisioner(sshClient, k8sVersion, kp.Slim)
+		if err = provisionK8sOpt.Exec(); err != nil {
+			return err
+		}
+	}
+
+	_ = sshClient.Command("sudo shutdown now -h")
+
+	_, err = docker.Exec(kp.Docker, kp.nodeContainer(kp.Version, nodeName), []string{"rm", "/ssh_ready"}, io.Discard)
+	if err != nil {
+		return err
+	}
+
+	logrus.Info("waiting for the node to stop")
+	okChan, errChan := kp.Docker.ContainerWait(ctx, kp.nodeContainer(kp.Version, nodeName), container.WaitConditionNotRunning)
+	select {
+	case <-okChan:
+	case err := <-errChan:
+		if err != nil {
+			return fmt.Errorf("waiting for the node to stop failed: %v", err)
+		}
+	}
+
+	if len(kp.AdditionalKernelArgs) > 0 {
+		dir, err := os.MkdirTemp("", "gocli")
+		if err != nil {
+			return fmt.Errorf("failed creating a temporary directory: %v", err)
+		}
+		defer os.RemoveAll(dir)
+		if err := os.WriteFile(filepath.Join(dir, "additional.kernel.args"), []byte(shellescape.QuoteCommand(kp.AdditionalKernelArgs)), 0666); err != nil {
+			return fmt.Errorf("failed creating additional.kernel.args file: %v", err)
+		}
+		if err := kp.copyDirectory(ctx, kp.Docker, node.ID, dir, "/"); err != nil {
+			return fmt.Errorf("failed copying additional kernel arguments into the container: %v", err)
+		}
+	}
+
+	logrus.Infof("Commiting the node as %s", target)
+	_, err = kp.Docker.ContainerCommit(ctx, node.ID, container.CommitOptions{
+		Reference: target,
+		Comment:   "PROVISION SUCCEEDED",
+		Author:    "gocli",
```

Comment:
Should the author be "The KubeVirt Team"?

Reply:
Yeah, we can change it